txpipe/lens_selector.py

```diff
@@ -0,0 +1,271 @@
+from .base_stage import PipelineStage
+from .data_types import YamlFile, TomographyCatalog, HDFFile, TextFile
+from .utils import LensNumberDensityStats
+import numpy as np
+import warnings
+
+class TXLensSelector(PipelineStage):
+    """
+    This pipeline stage selects objects to be used
+    as the lens sample for the galaxy clustering and
+    shear-position calibrations.
+    """
+
+    name='TXLensSelector'
+
+    inputs = [
+        ('calibration_table', TextFile),
+        ('photometry_catalog', HDFFile),
+    ]
+
+    outputs = [
+        ('lens_tomography_catalog', TomographyCatalog)
+    ]
+
+    config_options = {
+        'verbose': False,
+        'chunk_rows':10000,
+        'lens_zbin_edges':[float],
+        # Mag cuts
+        # Default photometry cuts based on the BOSS Galaxy Target Selection:                                                     
+        # http://www.sdss3.org/dr9/algorithms/boss_galaxy_ts.php                                           
+        'cperp_cut':0.2,
+        'r_cpar_cut':13.5,
+        'r_lo_cut':16.0,
+        'r_hi_cut':19.6,
+        'i_lo_cut':17.5,
+        'i_hi_cut':19.9,
+        'r_i_cut':2.0,
+        'random_seed': 42,
+    }
+
+    def run(self):
+        """
+        Run the analysis for this stage.
+        
+         - Collect the list of columns to read
+         - Create iterators to read chunks of those columns
+         - Loop through chunks:
+            - select objects for each bin
+            - write them out
+            - accumulate selection bias values
+         - Average the selection biases
+         - Write out biases and close the output
+        """
+        import astropy.table
+        import sklearn.ensemble
+
+        # Suppress some warnings from numpy that are not relevant
+        original_warning_settings = np.seterr(all='ignore')  
+
+        # The output file we will put the tomographic
+        # information into
+        output_file = self.setup_output()
+
+        # various config options
+        chunk_rows = self.config['chunk_rows']
+
+        phot_cols = ['i_mag','r_mag','g_mag', 'redshift_true']
```

Comment:
Can you print out a note here that we are cheating and using the true redshift?

Reply:
done